### PR TITLE
feat(COR-1354): add check of mime type on uploading assets to project

### PR DESCRIPTION
**Pull request by gheorghetutunaru**
- [ ] I opened a merge request on `kili` on a branch with the same name than the branch of the current pull request and forced security tests.
- [ ] I opened a pull request on the Node playground if the changes are breaking.

### CODE REVIEW

File: kili/mutations/asset/helpers.py [hunk 1]
```diff
@@ -74,12 +77,29 @@ def process_content(input_type: str,
...
     if input_type == 'FRAME' and json_content_array is None:
-        content_array = list(map(encode_object_if_not_url, content_array))
+        content_array = [encode_object_if_not_url(content, input_type) for content in content_array]
     return content_array
 
+def check_file_mime_type(content: str, input_type: str):
+    """
+    Returns true if the mime type of the file corresponds to the allowed mime types of the project
+    """
+    if input_type in ['IMAGE', 'FRAME', 'PDF']:
```
Comment (mDuval1) [inline kili/mutations/asset/helpers.py:90]: early return

```
if input_type not in ['IMAGE', 'FRAME', 'PDF']:
  return True
[...]
```

File: kili/mutations/asset/helpers.py [hunk 2]
```diff
@@ -74,12 +77,29 @@ def process_content(input_type: str,
...
+        mime_type = get_data_type(content.lower())
+        if mime_extensions_for_IV2[input_type] and mime_type:
+            correct_mime_type = mime_type in mime_extensions_for_IV2[input_type]
+            if not correct_mime_type:
+                print(f'File mime type for {content} is {mime_type} and does not correspond' \
+                    'to the type of the project. '\
+                    f'File mime type should be one of {mime_extensions_for_IV2[input_type]}')
+            return correct_mime_type
+        if not (mime_extensions_for_IV2[input_type] and mime_type):
```
Comment (mDuval1) [inline kili/mutations/asset/helpers.py:99]: The if is not necessary, you can return False directly

File: kili/mutations/asset/helpers.py [hunk 3]
```diff
@@ -74,12 +77,29 @@ def process_content(input_type: str,
...
             if (json_content_array is not None and json_content_array[i] is not None)
-                else encode_base64(content))
+            else (encode_base64(content) if check_file_mime_type(content, input_type) else None))
             for i, content in enumerate(content_array)]
     if input_type == 'FRAME' and json_content_array is None:
-        content_array = list(map(encode_object_if_not_url, content_array))
+        content_array = [encode_object_if_not_url(content, input_type) for content in content_array]
     return content_array
 
+def check_file_mime_type(content: str, input_type: str):
```
Comment (mDuval1) [inline kili/mutations/asset/helpers.py:86]: missing type return

File: kili/mutations/asset/helpers.py (hunk 2 above)
Comment (gheorghetutunaru) [reply kili/mutations/asset/helpers.py]: Also added as an early return

**Review by pmKili — APPROVED** [review]

**Review by mDuval1 — APPROVED** [review]

**Pull request merged**